### PR TITLE
Make logs collection more resilient to failures

### PR DESCRIPTION
**Pull request by rwx788**
We already use `script_run` in most of the places instead of
`assert_script_run` in case of logs collection to attempt gathering more
information even if parts of the system might be broken.
However, we use `upload_logs` and `script_output` calls without flags to
proceed on failure, which prevent further post_fail_hook steps from the
execution.

See [poo#77161](https://progress.opensuse.org/issues/77161).

### Verification runs
Here we can see that we even have a mechanism to recover network in trivial cases (I've added command to turn off the interface), but previously we would not even attempt it, as one of the upload commands will fail.
* [Before](https://openqa.suse.de/tests/5492920)
* [After](https://openqa.suse.de/tests/5492924)
* [Normal post fail hook execution](https://openqa.suse.de/tests/5490428)

### CODE REVIEW

**Review by jknphy — COMMENTED** [review]
LGTM

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -1308,9 +1308,9 @@ sub post_fail_hook {
     if (get_var('FULL_LVM_ENCRYPT') && get_var('LVM_THIN_LV')) {
         select_console 'root-console';
         my $lvmdump_regex = qr{/root/lvmdump-.*?-\d+\.tgz};
-        my $out           = script_output 'lvmdump';
+        my $out           = script_output('lvmdump', proceed_on_failure => 1);
         if ($out =~ /(?<lvmdump_gzip>$lvmdump_regex)/) {
-            upload_logs "$+{lvmdump_gzip}";
+            upload_logs("$+{lvmdump_gzip}", failok => 1);
         }
         $self->save_and_upload_log('lvm dumpconfig', '/tmp/lvm_dumpconf.out');
```
[the remark this answers is not in the document]
Comment (rwx788) [reply lib/opensusebasetest.pm:1318]: Not sure if I understand the point, `save_and_upload_log` has been changed too, see L98

[the remark this answers is not in the document]
Comment (jknphy) [reply lib/opensusebasetest.pm:1318]: missed that, thanks. all good! :)

[the remark this answers is not in the document]
Comment (rwx788) [reply lib/opensusebasetest.pm:1318]: No probs at all ;)

**Comment by okurz** [comment]
I wonder if we should make every call within the post_fail_hook non-fatal to try to continue but then again a lot more failed calls can be confusing to test reviewers as well, hm …

**Comment by rwx788** [comment]
> I wonder if we should make every call within the post_fail_hook non-fatal to try to continue but then again a lot more failed calls can be confusing to test reviewers as well, hm …

I also had same idea, but I believe that there are cases where we want to stop, if something is terribly wrong and we know that it's waste of time. E.g. in case of explicit `die`. From my experience, good use case is if we can multiple post fail hooks (parent one first and then other one), it would be ideal if we fail from the one, we stop execution of the first one and attempt the next one. It will provide granularity, as with the solution I've provided now we can spend more time and still not be able to collect any logs (risk is there). Consequently, I would leave a possibility to stop trying/skip some steps in case we are sure that we cannot collect any valuable information. For instance, do not execute multiple `coredumpctl` commands when we know that binary is not available.

**Pull request merged**